gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/map/DHMBase64Serializer.java

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright (c) 2010-2015 Pivotal Software, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package com.pivotal.gemfirexd.internal.engine.map;
+
+import com.google.common.hash.Hashing;
+
+/**
+ * Tries to compress key with base64 encoding assuming ascii characters only,
+ * in other words, assuming [a-z][A-Z][0-9] and [.*,- ] special characters only.
+ * <p/>
+ * note: one cannot have escape character '\' as well. only dot, asterik,
+ * hyphen, comma and space characters are allowed. Other characters appear as
+ * space and won't match different impls of {@code getHashCode} and {@code equals} methods.
+ */
+class DHMBase64Serializer implements DenseHashMapSerializer<String> {
```

Comment:
Why is Base64 being used? Its neither compact nor efficient. Why not simply use UTF8 encoding - String.getBytes(StandardCharsets.UTF_8) and corresponding constructor? It is both more compact and efficient than Base64.